frontend/src/v4/services/viewer/viewer.tsx

```diff
@@ -115,7 +115,10 @@ export class ViewerService {
 		unityHolder.style['pointer-events'] = 'all';
 
 		this.element.appendChild(this.viewer);
-		this.viewer.appendChild(unityHolder);
+
+		if (!this.viewer.hasChildNodes()) {
```

Comment:
You can probably move this check at the top of the method body and terminate the method if the condition holds true

Also, there seem to exists a method called `hasInstance` right before this method, probably you can that one